File: src/Vts/MonteCarlo/DataStructures/DetectorInputs/ROfRhoDetectorInput.cs
//using System;
//using System.Collections.Generic;
//using Vts.Common;

//namespace Vts.MonteCarlo
//{
//    /// <summary>
//    /// DetectorInput for R(r)
//    /// </summary>
//    public class ROfRhoDetectorInput : IDetectorInput
//    {
//        /// <summary>
//        /// constructor for reflectance as a function of rho detector input
//        /// </summary>
//        /// <param name="rho">rho binning</param>
//        /// <param name="name">detector name</param>
//        public ROfRhoDetectorInput(DoubleRange rho, String name)
//        {
//            TallyType = "ROfRho";
//            Name = name;
//            Rho = rho;
//        }
//        /// <summary>
//        /// constructor uses TallyType as name
//        /// </summary>
//        /// <param name="rho">rho binning</param>
//        public ROfRhoDetectorInput(DoubleRange rho) 
//            : this (rho, "ROfRho") {}

//        /// <summary>
//        /// Default constructor uses default rho bins
//        /// </summary>
//        public ROfRhoDetectorInput() 
//            : this(new DoubleRange(0.0, 10, 101),"ROfRho") {}

//        /// <summary>
//        /// detector tally identifier
//        /// </summary>
//        public string TallyType { get; set; }
//        /// <summary>
//        /// detector name, defaults to TallyType.ToString() but can be user specified
//        /// </summary>
//        public string Name { get; set; }
//        /// <summary>
//        /// detector rho binning
//        /// </summary>
//        public DoubleRange Rho { get; set; }
//    }
//}

File: matlab/vts_wrapper/vts/structures/MonteCarlo/DetectorInputs/ROfRhoDetectorInput.m
classdef ROfRhoDetectorInput < handle % deriving from handle allows us to keep a singleton around (reference based) - see Doug's post here: http://www.mathworks.com/matlabcentral/newsreader/view_thread/171344
    properties
        % detector tally identifier
        TallyType = 'ROfRho';
        % detector name, defaults to 'ROfRho' but can be user specified
        Name = 'ROfRho';
        % detector rho binning
        Rho = linspace(0, 10, 101);
    end
    
    methods
        % "constructor"
        function obj = ROfRhoDetectorInput(rho, name)
            if nargin == 2
                obj.Rho = rho;
                obj.Name = name;
            elseif nargin == 1
                obj.Rho = rho;
            end
        end
    end
    
    methods (Static)
        function input = FromInputNET(inputNET)
            input = ROfRhoDetectorInput();
            input.TallyType = char(inputNET.TallyType);
            input.Name = char(inputNET.Name);
            input.Rho = linspace(inputNET.Rho.Start, inputNET.Rho.Stop, inputNET.Rho.Count);
        end
        
        function inputNET = ToInputNET(input)
            inputNET = Vts.MonteCarlo.ROfRhoDetectorInput( ...
                Vts.Common.DoubleRange(input.Rho(1), input.Rho(end), length(input.Rho)), ...
                input.Name ...
                );
        end
    end
end